src/pumipic_adjacency.tpp

```diff
@@ -395,16 +395,25 @@ namespace pumipic {
     printf("Min area is: %.15f, Planned tol is %.15f\n", min_area, tol);
     return tol;
   }
-  
-  template <class ParticleType, typename Segment3d, typename SegmentInt>
-  bool search_mesh(o::Mesh& mesh, ParticleStructure<ParticleType>* ptcls,
+
+  template <class ParticleType, typename Segment3d, typename SegmentInt, typename Func>
+  bool particle_search(o::Mesh& mesh, ParticleStructure<ParticleType>* ptcls,
                    Segment3d x_ps_orig, Segment3d x_ps_tgt, SegmentInt pids,
                    o::Write<o::LO>& elem_ids,
                    bool requireIntersection,
```

Comment:
`requireIntersection` is specific to a particular type of behavior at element boundaries and should be removed from the function signature and put inside the functor instead.

Reply:
The `requireIntersection` flag is used inside the function here. For example, `find_exit_face` needs it. Please find it [here](https://github.com/SCOREC/pumi-pic/blob/87a7570f39fd2912b3b65938265edd901904f016/src/pumipic_adjacency.tpp#L420)

Reply:
I see what you mean, but in tracking this down I did find something else we need to address to decide what to do here.

In this code
https://github.com/SCOREC/pumi-pic/blob/87a7570f39fd2912b3b65938265edd901904f016/src/pumipic_adjacency.tpp#L490-L493

We fill up `lastExit` and `inter_points` arrays. Where, these correspond to the element face that the particle passes through using `find_exit_face`.

However, in `check_model_intersection` we fill in the `inter_faces` array. I *think* (but could be wrong) that `inter_faces` is supposed to be the faces that the particle passes through on the boundary.

If this is correct, then what's currently called `inter_faces` should not be in the search, but should be inside the functor.

And, the naming of `lastExit` and `inter_points` needs to be made consistent. The current naming is confusing since it seems like `inter_points` and `inter_faces` are referring to the same intersection but those are not matched.

Reply:
Yes, you are right. It should be in the functor. Search doesn't depend on it. You are correct about the naming. I had to write down what they do or how they are used.